### ISSUE TITLE
How do you run this?

### ISSUE THREAD

**Issue by RealStahl** [Nibre/HaloWwise#2]
Is says 

Usage:
 HaloWwise.exe <input.pck> <extraction path>
 HaloWwise.exe <input directory (recursive)> <extraction path>
  
  Where is the .exe?
  
  There is no .exe in the files

**Comment by JaneDough2022** [comment]
I also need help, I already built the program through Visual Studio. I don't know how to use it though, the program opens for a split moment and then it closes. 😅

**Comment by Medstar117** [comment]
> I also need help, I already built the program through Visual Studio. I don't know how to use it though, the program opens for a split moment and then it closes. 😅

You need to run it via the command prompt. Open a new command prompt, navigate to where HaloWwise.exe is located, then type the exe's name along with any parameters it requires. You could also make a batch script to do this for you.

**Comment by JaneDough2022** [comment]
> > I also need help, I already built the program through Visual Studio. I don't know how to use it though, the program opens for a split moment and then it closes. 😅
> 
> You need to run it via the command prompt. Open a new command prompt, navigate to where HaloWwise.exe is located, then type the exe's name along with any parameters it requires. You could also make a batch script to do this for you.

Yeah, I don't know how to make a batch script or do anything really after building the executable file. I'm very new to all this programming business, haha 😅. I only know how to build the code from Visual Studio. I only started doing all this about 8 days ago, all to just find certain sounds or voices within the files more easily.

**Comment by JaneDough2022** [comment]
Okay, I did the first step to extracting from the "soundvoice.pck" file.
Now I have multiple folders with the name "SoundBank (0xXXXXXXXX)" and a folder called "SoundFiles" with multiple 0xXXXXXXXX.wem files.
Now, I don't understand what the second part is "<input directory (recursive)>". Sorry if I don't understand, I'm new to the programming business.

**Comment by JaneDough2022** [comment]
> > I also need help, I already built the program through Visual Studio. I don't know how to use it though, the program opens for a split moment and then it closes. 😅
> 
> You need to run it via the command prompt. Open a new command prompt, navigate to where HaloWwise.exe is located, then type the exe's name along with any parameters it requires. You could also make a batch script to do this for you.

Okay, I did the first step to extracting from the "soundvoice.pck" file.
Now I have multiple folders with the name "SoundBank (0xXXXXXXXX)" and a folder called "SoundFiles" with multiple 0xXXXXXXXX.wem files.
Now, I don't understand what the second part is "<input directory (recursive)>". Sorry if I don't understand, I'm new to the programming business.

**Comment by Medstar117** [comment]
> > > I also need help, I already built the program through Visual Studio. I don't know how to use it though, the program opens for a split moment and then it closes. 😅
> > 
> > 
> > You need to run it via the command prompt. Open a new command prompt, navigate to where HaloWwise.exe is located, then type the exe's name along with any parameters it requires. You could also make a batch script to do this for you.
> 
> Okay, I did the first step to extracting from the "soundvoice.pck" file. Now I have multiple folders with the name "SoundBank (0xXXXXXXXX)" and a folder called "SoundFiles" with multiple 0xXXXXXXXX.wem files. Now, I don't understand what the second part is "<input directory (recursive)>". Sorry if I don't understand, I'm new to the programming business.

The recursive bit is like this:

Let's say you have two .pck files you're working with. You put the two files in a new folder. You then give HaloWwise the full path to this new folder. HaloWwise will then extract both .pck files.

So basically, it's just another input method, not another step you need to do. "Recursive" just means it will do the same operation to multiple files within the same directory.

**Comment by JaneDough2022** [comment]
> > > > I also need help, I already built the program through Visual Studio. I don't know how to use it though, the program opens for a split moment and then it closes. 😅
> > > 
> > > 
> > > You need to run it via the command prompt. Open a new command prompt, navigate to where HaloWwise.exe is located, then type the exe's name along with any parameters it requires. You could also make a batch script to do this for you.
> > 
> > 
> > Okay, I did the first step to extracting from the "soundvoice.pck" file. Now I have multiple folders with the name "SoundBank (0xXXXXXXXX)" and a folder called "SoundFiles" with multiple 0xXXXXXXXX.wem files. Now, I don't understand what the second part is "<input directory (recursive)>". Sorry if I don't understand, I'm new to the programming business.
> 
> The recursive bit is like this:
> 
> Let's say you have two .pck files you're working with. You put the two files in a new folder. You then give HaloWwise the full path to this new folder. HaloWwise will then extract both .pck files.
> 
> So basically, it's just another input method, not another step you need to do. "Recursive" just means it will do the same operation to multiple files within the same directory.

Okay, thanks for making it more friendly to understand that part of the process. 
I also have a question, it's about the names of the SoundBank 0xXXXXXXXX folders inside of the Soundbank.pck and Soundvoice.pck I extracted. Is there a file I'm missing? So I can know what the names are of these 0xXXXXXXXX inside both pcks? I'm using the Reclaimer tool and using the Module Viewer, viewing the "globals-rtx-1.module" file from the Halo 5 Forge dump as a reference to the names of the folders, but I guess its a more complicated process to this I'm guessing?
